library/core/src/ffi/va_list.rs

```diff
@@ -94,97 +113,30 @@ crate::cfg_select! {
     // - apple aarch64 (see https://github.com/rust-lang/rust/pull/56599)
     // - windows
     // - uefi
-    // - any other target for which we don't specify the `VaListImpl` above
+    // - any other target for which we don't specify the `VaList` above
     //
     // In this implementation the `va_list` type is just an alias for an opaque pointer.
     // That pointer is probably just the next variadic argument on the caller's stack.
     _ => {
         /// Basic implementation of a `va_list`.
         #[repr(transparent)]
         #[lang = "va_list"]
-        pub struct VaListImpl<'f> {
+        pub struct VaList<'f> {
             ptr: *mut c_void,
 
-            // Invariant over `'f`, so each `VaListImpl<'f>` object is tied to
+            // Invariant over `'f`, so each `VaList<'f>` object is tied to
             // the region of the function it's defined in
             _marker: PhantomInvariantLifetime<'f>,
         }
 
-        impl<'f> fmt::Debug for VaListImpl<'f> {
+        impl<'f> fmt::Debug for VaList<'f> {
             fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                write!(f, "va_list* {:p}", self.ptr)
+                f.debug_tuple("VaList").field(&self.ptr).finish()
```

Comment:
`Debug` is derived for all the other implementations, we can do that here too now right?